docs/architecture/cloud-native/rest-grpc.md

```diff
@@ -1,69 +1,347 @@
+gRPC embraces an open-source technology called [Protocol Buffers](https://developers.google.com/protocol-buffers/docs/overview). They provide a highly efficient and platform-neutral serialization format for serializing structured messages that services send to each other. Using a cross-platform Interface Definition Language (IDL), developers define a service contract for each microservice. The contract, implemented as a text-based '.proto' file, describes the methods, inputs, and outputs for each service. The same contract file can be used for gRPC clients and services built on different development platforms.
 
-Many object serialization techniques reflect across the structure of data objects at run-time. Protobuf requires you to define the structure up front with a platform-agnostic language (Protocol Buffer Language). Each definition is stored in a ".proto" file. Then using Protobuf compiler, "Proton," you generate client and server code for any of the supported platforms. The generated code is optimized for fast serialization/deserialization of data. At runtime, each message is wrapped in the strongly-typed service contract and serialized in a standard Protobuf representation.
+Using the proto file," the Protobuf compiler, "Protoc," generates both client and service code for your target platform. The code includes the following components:
+
+- A strongly-typed DTO (Data Transfer Object) shared by the client and service.
+- A strongly-typed base class with the required network plumbing that the remote gRPC service can inherit and extend.
+- A client stub that contains the required plumbing to invoke the remote gRPC service.
+
+At runtime, each message is serialized as a standard Protobuf representation and exchanged between the client and remote service. Unlike JSON or XML, Protobuf messages are serialized as compiled binary bytes.
+
+The book, [gRPC for WCF Developers](https://docs.microsoft.com/dotnet/architecture/grpc-for-wcf-developers/), available from the Microsoft Architecture Site, provides in-depth coverage of gRPC and Protocol Buffers.
 
 ## gRPC support in .NET
 
-The Microsoft .NET Core framework 3.0 includes tooling and native support for gRPC. Figure 4-20 shows the Visual Studio 2019 template that scaffolds a gRPC skeleton project for a gRPC service. Note how .NET Core supports the Windows, Linux, and macOS platforms.
+Microsoft .NET Core 3.0 includes tooling and native support for gRPC. gRPC has been seamlessly integrated into the 3.0 framework with support for endpoint routing, built-in IoC, and logging. The open-source Kestrel web server supports HTTP/2 connections. Figure 4-20 shows a Visual Studio 2019 template that scaffolds a skeleton project for a gRPC service. Note how .NET Core fully supports the Windows, Linux, and macOS platforms.
```

Comment:
Rephrase this entire para. The use of the word `framework` is confusing in the context of .NET Core. Use `.NET Core 3.0 SDK`
Also, 
- mention gRPC is supported in .NET Core 3.0 and **later**. 
- Visual Studio 2019 version 16.3 or later, with the ASP.NET and web development workload installed.
- support in dotnet CLI 

The sentence is confusing:

> Note how .NET Core fully supports the Windows, Linux, and macOS platforms.

Reply:
Refactored paragraph